### PR TITLE
Use completing-read in interactive prompt for ruby version

### PR DESCRIPTION
**Pull request by til**
One more 😄 .

This way one does not have to guess or remember which rubies are available. Works for me with helm, and also without helm.

### CODE REVIEW

File: chruby.el [hunk 1]
```diff
@@ -174,6 +174,10 @@ ruby version, and the gem path"
     (if version-file-path (chruby-use (chruby--read-version-from-file version-file-path))
       (message "[chruby] could not locate .ruby-version"))))
 
+(defun chruby--available-names ()
+  "list of available ruby names without path, higher versions sorted first"
+  (reverse (cl-mapcar 'chruby-util-basename (chruby-rubies))))
```
Comment (plexus) [inline chruby.el]: I don't think you need `cl-mapcar` here, regular `mapcar` should do.

From [the manual](https://www.gnu.org/software/emacs/manual/html_node/cl/Mapping-over-Sequences.html)

>  Given a single seq argument it is equivalent to mapcar; given n sequences, it calls the function with the first elements of each of the sequences as the n arguments to yield the first element of the result list, then with the second elements, and so on.

**Comment by til** [comment]
I saw that cl-mapcar is used here as well although mapcar is sufficent
(I think):
https://github.com/plexus/chruby.el/blob/master/chruby.el#L78-L81 and
followed.

What's the general recommendation - use cl-* functions only when needed?

Anyway I'll change it in the PR, hold on.

**Comment by plexus** [comment]
Thanks for the PR! Definitely an improvement. I now remember why these functions weren't documented. They were contributed by someone else, and I guess that person didn't update the docs. I had forgotten they even existed :)

Yeah, in general I'd say only use the `cl` functions when there's no regular alternative. But it doesn't matter so much to be honest. I just added `cl-lib` as an explicit dependency now, that wasn't the case yet.

**Comment by til** [comment]
Thanks for the feedback! Pushed a change to use mapcar.

**Comment by plexus** [comment]
cake

yay!

**Pull request merged**